src/org/etools/j1939_84/controllers/part1/Part01Step09Controller.java

```diff
@@ -134,26 +139,26 @@ protected void run() throws Throwable {
         // a. Warn if the serial number field (SPN 588) from any function 0
         // device is less than 8 characters long.
         if (serialNumber.length() < 8) {
-            addWarning(PART_NUMBER,
-                    STEP_NUMBER,
+            addWarning(getPartNumber(),
+                    getStepNumber(),
                     "6.1.9.3.a Serial number field (SPN 588) from any function 0 device is less than 8 characters long");
         }
         // b. Warn if the make field (SPN 586) is longer than 5 ASCII characters.
         if (!make.isEmpty() && make.length() > 5) {
-            addWarning(PART_NUMBER,
-                    STEP_NUMBER,
+            addWarning(getPartNumber(),
+                    getStepNumber(),
                     "6.1.9.3.b Make field (SPN 586) is longer than 5 ASCII characters");
         }
         // c. Warn if the make field (SPN 586) is less than 2 ASCII characters.
         if (make.length() == 1) {
-            addWarning(PART_NUMBER,
-                    STEP_NUMBER,
+            addWarning(getPartNumber(),
```

Comment:
you can use the addWarning() method that doesn't need the part number or step number passed in and clean this up a bit.